Jenkinsfile

```
@@ -0,0 +1,30 @@
+pipeline {
+  agent any
+  stages {
+    stage('Validate the pushed code') {
+      steps {
+        sh 'stack exec --package=hlint -- hlint --git'
+      }
+    }
+    stage('Run tests') {
+      steps {
+        timestamps () {
+          sh 'stack test --fast --test-arguments "--xml junit.xml --display t"'
```

```suggestion
-          sh 'stack test --fast --test-arguments "--xml junit.xml --display t"'
+          sh 'env AMC_LIBRARY_PATH=$PWD/lib/ stack --no-terminal test --fast --ghc-options "-Werror -fmax-pmcheck-iterations=5000000" --test-arguments "--xml junit.xml --display t --hedgehog-tests 5000 --timeout 80'
```

Maybe? I'd also be inclined to split this into install deps, build, and then test, like we do for Travis.

---

the deps are shared though